src/block-management/create-block-and-log-streamer.js

```diff
@@ -57,17 +56,30 @@ function createBlockAndLogStreamer(configuration, transport) {
     });
 
     internalState.setState({ blockAndLogStreamer: blockAndLogStreamer, blockNotifier: blockNotifier });
+
+    callback = callback ? callback : function (e) { if (e) console.log(e); };
     function subscribeToBlockNotifier() {
       blockNotifier.subscribe(function (block) {
         blockAndLogStreamer.reconcileNewBlockCallbackStyle(block, function (err) { if (err) return console.error(err); });
       });
+
+      callback(null);
     }
 
     if (typeof configuration.startingBlockNumber !== "undefined") {
-      var block = dispatch(eth_getBlockByNumber([configuration.startingBlockNumber, false]));
-      blockAndLogStreamer.reconcileNewBlockCallbackStyle(block, function (err) {
-        if (err) return console.error(err);
-        subscribeToBlockNotifier();
+      transport.getBlockByNumber(configuration.startingBlockNumber, function (err, block) {
+        if (err) {
+          return console.log(err);
+        }
+
+        blockAndLogStreamer.reconcileNewBlockCallbackStyle(block, function (err) {
+          if (err) {
+            if (callback) return callback(err);
+            return console.log(err);
```

Comment:
The callback is always defined now, so these two lines can be simplified to just `return callback(err)`.